app/src/parser_print_txn.c

```diff
@@ -183,6 +185,217 @@ static parser_error_t printTransferTxn( const parser_context_t *ctx,
+    // Get pointer to the spends
+    bytes_t spend = ctx->tx_obj->transaction.sections.maspBuilder.builder.sapling_builder.spends;
+
+    // For each spend, we will print 3 items, if we have more then 3 items we need to rebase the display Idx to keep printing the spends
+    if(n_send_items > 3 ) {
+        if(displayIdx >= 4 && displayIdx <= n_send_items) {
+            displayIdx = displayIdx % 3;
+            displayIdx = displayIdx == 0 ? 3 : displayIdx;
+            if (displayIdx == 1 && pageIdx == 0) {
+                // If displayIdx was rebase to first item to be printed, we need to increment the spend index
+                spend_index ++;
+            }
+        } else if(displayIdx > n_send_items) { // If we are done printing the spends, we need to keep rebasing the display idx so we can enter the corrent case
+            displayIdx -= (n_send_items - 3);
+            spend_index = 0;
+        }
+        // Get new spend pointer
+        getSpendfromIndex(spend_index, &spend);
+    }
+
+    // Get pointer to the outputs
+    bytes_t out = ctx->tx_obj->transaction.sections.maspBuilder.builder.sapling_builder.outputs;
+
+    // For each output, we will print 3 items, if we have more then 3 items we need to rebase the display Idx to keep printing the outputs
+    if(n_dest_items > 3 ) {
+        if(displayIdx >= 7 && displayIdx <= n_send_items + n_dest_items && orig_idx <= (n_send_items + n_dest_items)) {
+            displayIdx = (displayIdx % 6) + 3;
+            if (displayIdx == 4 && pageIdx == 0) {
+                // If displayIdx was rebase to first item to be printed, we need to increment the out index
+                out_index ++;
+            }
+        } else if(displayIdx > n_dest_items) {
+            displayIdx -= (n_dest_items - 3);
+            out_index = 0;
+        }
+        // Get new output pointer
+        getOutputfromIndex(out_index, &out);
+    }
+
+    // Check if Memo needs to be printed or not
+    const bool hasMemo = ctx->tx_obj->transaction.header.memoSection != NULL;
+    if (displayIdx >= 7 && !hasMemo) {
+        displayIdx++;
+    }
+
+    char tmp_buf[500] = {0};
+    uint8_t tmp_amount[32] = {0};
+    bytes_t amount_bytes = {tmp_amount, 32};
+    const uint8_t *amount = {0};
+
+    switch (displayIdx) {
+        case 0:
+            snprintf(outKey, outKeyLen, "Type");
+            snprintf(outVal, outValLen, "Transfer");
+            if (app_mode_expert()) {
+                CHECK_ERROR(printCodeHash(&ctx->tx_obj->transaction.sections.code, outKey, outKeyLen,
+                                          outVal, outValLen, pageIdx, pageCount))
+            }
+            break;
+        case 1:
+            snprintf(outKey, outKeyLen, "Sender");
+            if (ctx->tx_obj->transfer.source_address.tag != 2) {
+                CHECK_ERROR(printAddressAlt(&ctx->tx_obj->transfer.source_address, outVal, outValLen, pageIdx, pageCount))
+            } else {
+                const char *hrp = "zvknam";
```

Comment:
mainnet and testnet hrp are different. Probably we need to move the encode function to cryptohelper that already consider the different prefixes for transparent pubkey and addresses